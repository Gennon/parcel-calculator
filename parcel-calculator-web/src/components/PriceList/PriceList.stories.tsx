import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { PriceList } from './PriceList';

export default {
  title: 'ParcelCalculator/PriceList',
  component: PriceList,
  argTypes: {},
} as ComponentMeta<typeof PriceList>;

const Template: ComponentStory<typeof PriceList> = args => (
  <PriceList {...args} />
);

export const Default = Template.bind({});
