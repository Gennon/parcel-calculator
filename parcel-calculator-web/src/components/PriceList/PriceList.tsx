import {
  Table,
  TableCaption,
  TableContainer,
  Tbody,
  Td,
  Tfoot,
  Th,
  Thead,
  Tr,
} from '@chakra-ui/react';
import { FC } from 'react';

export const PriceList: FC = () => {
  const pricePerParcel = 50;
  const pricePerKg = 5;
  const initialPrice = 10;

  const calculatePrice = (numberOfParcels: number, kg: number) => {
    return (
      numberOfParcels * pricePerParcel +
      numberOfParcels * pricePerKg * (kg - 1) +
      initialPrice
    );
  };

  const headers = [1, 2, 3, 4, 5];
  const rows = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];

  return (
    <TableContainer>
      <Table variant="striped">
        <TableCaption>Price for #parcels in kg</TableCaption>
        <Thead>
          <Tr>
            <Th>kg</Th>
            {headers.map(header => (
              <Th key={header}>#{header}</Th>
            ))}
          </Tr>
        </Thead>
        <Tbody>
          {rows.map(row => (
            <Tr key={row}>
              <Td>{row}</Td>
              {headers.map(header => (
                <Td key={header}>{calculatePrice(header, row)},-</Td>
              ))}
            </Tr>
          ))}
        </Tbody>
        <Tfoot>
          <Tr>
            <Th>kg</Th>
            {headers.map(header => (
              <Th key={header}>#{header}</Th>
            ))}
          </Tr>
        </Tfoot>
      </Table>
    </TableContainer>
  );
};
