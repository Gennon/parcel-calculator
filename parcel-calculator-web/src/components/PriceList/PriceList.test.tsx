import React from 'react';
import { render, screen } from '@testing-library/react';
import { PriceList } from './PriceList';

test('it exists', () => {
  const { container } = render(<PriceList />);
  expect(container).not.toBeNull();
});

test('has correct structure', () => {
  const { asFragment } = render(<PriceList />);
  expect(asFragment()).toMatchSnapshot();
});
