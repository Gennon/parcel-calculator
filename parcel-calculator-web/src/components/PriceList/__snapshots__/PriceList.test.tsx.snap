// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has correct structure 1`] = `
<DocumentFragment>
  <div
    class="chakra-table__container css-zipzvv"
  >
    <table
      class="chakra-table css-0"
    >
      <caption
        class="css-18i7bby"
      >
        Price for #parcels in kg
      </caption>
      <thead
        class="css-0"
      >
        <tr
          class="css-0"
        >
          <th
            class="css-0"
          >
            kg
          </th>
          <th
            class="css-0"
          >
            #1
          </th>
          <th
            class="css-0"
          >
            #2
          </th>
          <th
            class="css-0"
          >
            #3
          </th>
          <th
            class="css-0"
          >
            #4
          </th>
          <th
            class="css-0"
          >
            #5
          </th>
        </tr>
      </thead>
      <tbody
        class="css-0"
      >
        <tr
          class="css-0"
        >
          <td
            class="css-0"
          >
            1
          </td>
          <td
            class="css-0"
          >
            60,-
          </td>
          <td
            class="css-0"
          >
            110,-
          </td>
          <td
            class="css-0"
          >
            160,-
          </td>
          <td
            class="css-0"
          >
            210,-
          </td>
          <td
            class="css-0"
          >
            260,-
          </td>
        </tr>
        <tr
          class="css-0"
        >
          <td
            class="css-0"
          >
            2
          </td>
          <td
            class="css-0"
          >
            65,-
          </td>
          <td
            class="css-0"
          >
            120,-
          </td>
          <td
            class="css-0"
          >
            175,-
          </td>
          <td
            class="css-0"
          >
            230,-
          </td>
          <td
            class="css-0"
          >
            285,-
          </td>
        </tr>
        <tr
          class="css-0"
        >
          <td
            class="css-0"
          >
            3
          </td>
          <td
            class="css-0"
          >
            70,-
          </td>
          <td
            class="css-0"
          >
            130,-
          </td>
          <td
            class="css-0"
          >
            190,-
          </td>
          <td
            class="css-0"
          >
            250,-
          </td>
          <td
            class="css-0"
          >
            310,-
          </td>
        </tr>
        <tr
          class="css-0"
        >
          <td
            class="css-0"
          >
            4
          </td>
          <td
            class="css-0"
          >
            75,-
          </td>
          <td
            class="css-0"
          >
            140,-
          </td>
          <td
            class="css-0"
          >
            205,-
          </td>
          <td
            class="css-0"
          >
            270,-
          </td>
          <td
            class="css-0"
          >
            335,-
          </td>
        </tr>
        <tr
          class="css-0"
        >
          <td
            class="css-0"
          >
            5
          </td>
          <td
            class="css-0"
          >
            80,-
          </td>
          <td
            class="css-0"
          >
            150,-
          </td>
          <td
            class="css-0"
          >
            220,-
          </td>
          <td
            class="css-0"
          >
            290,-
          </td>
          <td
            class="css-0"
          >
            360,-
          </td>
        </tr>
        <tr
          class="css-0"
        >
          <td
            class="css-0"
          >
            6
          </td>
          <td
            class="css-0"
          >
            85,-
          </td>
          <td
            class="css-0"
          >
            160,-
          </td>
          <td
            class="css-0"
          >
            235,-
          </td>
          <td
            class="css-0"
          >
            310,-
          </td>
          <td
            class="css-0"
          >
            385,-
          </td>
        </tr>
        <tr
          class="css-0"
        >
          <td
            class="css-0"
          >
            7
          </td>
          <td
            class="css-0"
          >
            90,-
          </td>
          <td
            class="css-0"
          >
            170,-
          </td>
          <td
            class="css-0"
          >
            250,-
          </td>
          <td
            class="css-0"
          >
            330,-
          </td>
          <td
            class="css-0"
          >
            410,-
          </td>
        </tr>
        <tr
          class="css-0"
        >
          <td
            class="css-0"
          >
            8
          </td>
          <td
            class="css-0"
          >
            95,-
          </td>
          <td
            class="css-0"
          >
            180,-
          </td>
          <td
            class="css-0"
          >
            265,-
          </td>
          <td
            class="css-0"
          >
            350,-
          </td>
          <td
            class="css-0"
          >
            435,-
          </td>
        </tr>
        <tr
          class="css-0"
        >
          <td
            class="css-0"
          >
            9
          </td>
          <td
            class="css-0"
          >
            100,-
          </td>
          <td
            class="css-0"
          >
            190,-
          </td>
          <td
            class="css-0"
          >
            280,-
          </td>
          <td
            class="css-0"
          >
            370,-
          </td>
          <td
            class="css-0"
          >
            460,-
          </td>
        </tr>
        <tr
          class="css-0"
        >
          <td
            class="css-0"
          >
            10
          </td>
          <td
            class="css-0"
          >
            105,-
          </td>
          <td
            class="css-0"
          >
            200,-
          </td>
          <td
            class="css-0"
          >
            295,-
          </td>
          <td
            class="css-0"
          >
            390,-
          </td>
          <td
            class="css-0"
          >
            485,-
          </td>
        </tr>
      </tbody>
      <tfoot
        class="css-0"
      >
        <tr
          class="css-0"
        >
          <th
            class="css-0"
          >
            #1
          </th>
          <th
            class="css-0"
          >
            #2
          </th>
          <th
            class="css-0"
          >
            #3
          </th>
          <th
            class="css-0"
          >
            #4
          </th>
          <th
            class="css-0"
          >
            #5
          </th>
        </tr>
      </tfoot>
    </table>
  </div>
</DocumentFragment>
`;
